src/event/model/event.rs

```diff
@@ -91,9 +110,19 @@ impl Event {
             "entity/link/remove" => Event::RemoveEntityLink(RemoveEntityLink::build(e)),
             "entity/revision/add" => Event::CreateEntityRevision(CreateEntityRevision::build(e)),
             "entity/revision/checkout" => {
-                let repository_id = fetch_parameter_uuid_id(e.id, "repository", &pool).await;
+                let repository_id = fetch_parameter_uuid_id(e.id, "repository", &pool)
+                    .await
+                    .unwrap();
                 let reason = fetch_parameter_string(e.id, "reason", &pool).await;
-                Event::CheckoutRevision(CheckoutRevision::build(e, repository_id.unwrap(), reason))
+                let revision_id = abstract_event.object_id;
+                Event::CheckoutRevision(CheckoutRevision {
+                    abstract_event,
+
+                    __typename: String::from("CheckoutRevisionNotificationEvent"),
+                    repository_id,
+                    revision_id,
+                    reason,
+                })
```

Comment:
It probably makes sense to outsource that into `checkout_revision`. If you do that, though, I'd prefer the pattern we used for the uuid model instead, i.e. instead of fetching the type-specific data here, let the type handle those themselves. So, for example:
```rust
impl CheckoutRevision {
    // new is the idiomatic name for a constructor-like function
    pub async fn new(abstract_event: AbstractEvent, pool) -> Result<Self> {
        let reason = Event::fetch_parameter_string(abstract_event.id, "reason", &pool).await;
    }
}
```